mlrun/runtimes/databricks/databricks.py

```diff
@@ -19,31 +19,42 @@
 from mlrun.runtimes.kubejob import KubejobRuntime
 
 
+
 class DatabricksRuntime(KubejobRuntime):
     kind = "databricks"
     _is_remote = True
 
-    def get_internal_code(self, runobj: RunObject):
+    def get_internal_parameters(self, runobj: RunObject):
         """
         Return the internal function code.
         """
+        task_parameters = runobj.spec.parameters.get("task_parameters", {})
+        if "internal_handler" in task_parameters:
+            internal_handler = task_parameters["internal_handler"]
+        else:
+            internal_handler = runobj.spec.handler or ""
         encoded_code = (
             self.spec.build.functionSourceCode if hasattr(self.spec, "build") else None
         )
+        if not encoded_code:
+            return ""
         decoded_code = b64decode(encoded_code).decode("utf-8")
         code = _databricks_script_code + decoded_code
-        if runobj.spec.handler:
-            code += f"\n{runobj.spec.handler}(**handler_arguments)\n"
+        if internal_handler:
+            code += f"\n{internal_handler}(**handler_arguments)\n"
         code = b64encode(code.encode("utf-8")).decode("utf-8")
-        return code
+        return code, internal_handler
 
     def _pre_run(self, runspec: RunObject, execution):
-        internal_code = self.get_internal_code(runspec)
+        internal_code, internal_handler = self.get_internal_parameters(runspec)
         if internal_code:
             task_parameters = runspec.spec.parameters.get("task_parameters", {})
-            task_parameters["spark_app_code"] = self.get_internal_code(runspec)
+            task_parameters["spark_app_code"] = internal_code
+            if internal_handler:
```

Comment:
Consider renaming `internal_handler` to `user_handler` or `original_handler` or something else that's more informative.